connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTest.java

```diff
@@ -1819,16 +1825,25 @@ public void testGetSinkConnectorOffsetsAdminClientAsynchronousError() {
         verifyKafkaClusterId();
     }
 
-    @SuppressWarnings("unchecked")
     private void mockAdminListConsumerGroupOffsets(Admin admin, Map<TopicPartition, OffsetAndMetadata> consumerGroupOffsets, Exception e) {
+        mockAdminListConsumerGroupOffsets(admin, consumerGroupOffsets, e, null, 0);
+    }
+
+    private void mockAdminListConsumerGroupOffsets(Admin admin, Map<TopicPartition, OffsetAndMetadata> consumerGroupOffsets, Exception e, Time time, long delayMs) {
         ListConsumerGroupOffsetsResult result = mock(ListConsumerGroupOffsetsResult.class);
         when(admin.listConsumerGroupOffsets(anyString(), any(ListConsumerGroupOffsetsOptions.class))).thenReturn(result);
-        KafkaFuture<Map<TopicPartition, OffsetAndMetadata>> adminFuture = mock(KafkaFuture.class);
-        when(result.partitionsToOffsetAndMetadata()).thenReturn(adminFuture);
-        when(adminFuture.whenComplete(any())).thenAnswer(invocation -> {
-            ((KafkaFuture.BiConsumer<Map<TopicPartition, OffsetAndMetadata>, Throwable>) invocation.getArgument(0))
-                    .accept(consumerGroupOffsets, e);
-            return null;
+        KafkaFutureImpl<Map<TopicPartition, OffsetAndMetadata>> adminFuture = new KafkaFutureImpl<>();
+        if (e != null) {
+            adminFuture.completeExceptionally(e);
+        } else {
+            adminFuture.complete(consumerGroupOffsets);
+        }
+        when(result.partitionsToOffsetAndMetadata()).thenAnswer(invocation -> {
+            if (time == null) {
+                return adminFuture;
+            }
+            time.sleep(delayMs);
+            return adminFuture;
```

Comment:
Nit: can simplify this
```suggestion
            if (time != null)
                time.sleep(delayMs);
```